classes/class-wc-connect-service-settings-store.php

```diff
@@ -206,7 +208,49 @@ public function get_enabled_services() {
 
 			usort( $enabled_services, array( $this, 'sort_services' ) );
 			return $enabled_services;
+		}
+
+		/**
+		 * Checks if the shipping method ids have been migrated to the "wc_services_*" format and migrates them
+		 */
+		public function migrate_legacy_services() {
+			if ( WC_Connect_Options::get_option( 'shipping_methods_migrated', false ) //check if the method have already been migrated
+				|| ! $this->service_schemas_store->fetch_service_schemas_from_connect_server() //ensure the latest schemas are fetched
+				|| ! $this->service_schemas_store->get_all_shipping_method_ids() ) { //verify that the schemas contain the method_id field
+				return;
+			}
+
+			global $wpdb;
+
+			//old services used the id field instead of method_id
+			$shipping_service_ids = $this->service_schemas_store->get_all_service_ids_of_type( 'shipping' );
+			$legacy_services = $this->get_enabled_services_by_ids( $shipping_service_ids );
+
+			foreach ( $legacy_services as $legacy_service ) {
+				$service_id = $legacy_service->method_id;
+				$instance_id = $legacy_service->instance_id;
+				$service_schema = $this->service_schemas_store->get_service_schema_by_id( $service_id );
+				$service_settings = $this->get_service_settings( $service_id, $instance_id );
+				if ( ! is_object( $service_settings ) ) {
+					continue;
+				}
+
+				$new_method_id = $service_schema->method_id;
+
+				$query = $wpdb->prepare(
+					"UPDATE {$wpdb->prefix}woocommerce_shipping_zone_methods " .
+					"SET method_id=%s " .
+					"WHERE instance_id=%s AND method_id=%s",
+					$new_method_id, $instance_id, $service_id);
+				$wpdb->query( $query );
```

Comment:
Can [`$wpdb->update()`](https://developer.wordpress.org/reference/classes/wpdb/update/) be used here instead?